### PR TITLE
Move the CodeID type to actors/abi and convert to a real CID

### PR DESCRIPTION
**Pull request by anorth**
~Based on #794~

### CODE REVIEW

File: src/actors/builtin/codes.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+var SystemActorCodeID abi.ActorCodeID
+var InitActorCodeID abi.ActorCodeID
+var CronActorCodeID abi.ActorCodeID
+var AccountActorCodeID abi.ActorCodeID
+var StoragePowerActorCodeID abi.ActorCodeID
+var StorageMinerActorCodeID abi.ActorCodeID
+var StorageMarketActorCodeID abi.ActorCodeID
+var PaymentChannelActorCodeID abi.ActorCodeID
+var MultisigActorCodeID abi.ActorCodeID
+
```
Comment (zixuanzh) [inline src/actors/builtin/codes.go]: do we need to include BurntFundActor and RewardActor?

File: src/actors/builtin/codes.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	SystemActorCodeID = makeBuiltin("fil/1/system")
+	InitActorCodeID = makeBuiltin("fil/1/init")
+	CronActorCodeID = makeBuiltin("fil/1/cron")
+	AccountActorCodeID = makeBuiltin("fil/1/account")
+	StoragePowerActorCodeID = makeBuiltin("fil/1/storagepower")
+	StorageMinerActorCodeID = makeBuiltin("fil/1/storageminer")
+	StorageMarketActorCodeID = makeBuiltin("fil/1/storagemarket")
+	PaymentChannelActorCodeID = makeBuiltin("fil/1/paymentchannel")
+	MultisigActorCodeID = makeBuiltin("fil/1/multisig")
```
Comment (zixuanzh) [inline src/actors/builtin/codes.go]: ditto

File: src/systems/filecoin_vm/interpreter/vm_registry.go [hunk 3]
```diff
@@ -67,11 +56,11 @@ func _registerBuiltinActors() {
...
-	RegisterActor(InitActorCodeID, &initact.InitActorCode_I{})
-	RegisterActor(CronActorCodeID, cron)
-	RegisterActor(AccountActorCodeID, &accact.AccountActorCode_I{})
-	RegisterActor(StoragePowerActorCodeID, &spowact.StoragePowerActorCode_I{})
-	RegisterActor(StorageMarketActorCodeID, &smarkact.StorageMarketActorCode_I{})
+	RegisterActor(builtin.InitActorCodeID, &initact.InitActorCode_I{})
+	RegisterActor(builtin.CronActorCodeID, cron)
+	RegisterActor(builtin.AccountActorCodeID, &accact.AccountActorCode_I{})
+	RegisterActor(builtin.StoragePowerActorCodeID, &spowact.StoragePowerActorCode_I{})
+	RegisterActor(builtin.StorageMarketActorCodeID, &smarkact.StorageMarketActorCode_I{})
```
Comment (zixuanzh) [inline src/systems/filecoin_vm/interpreter/vm_registry.go]: ditto

File: src/actors/builtin/codes.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// Tests whether a CID is a builtin actor code ID.
+func IsBuiltinCodeID(c abi.ActorCodeID) bool {
```
Comment (zixuanzh) [inline src/actors/builtin/codes.go]: this function is never called and type abi.ActorCodeID can be any CID

File: src/actors/builtin/codes.go (hunk 1 above)
Comment (anorth) [reply src/actors/builtin/codes.go]: BurntFunds is just an account. Will add Reward

File: src/systems/filecoin_vm/interpreter/vm_registry.go (hunk 3 above)
Comment (anorth) [reply src/systems/filecoin_vm/interpreter/vm_registry.go]: There are many missing here, but I'm just trying to extract things, not fix them right now.

File: src/actors/builtin/codes.go (hunk 4 above)
Comment (anorth) [reply src/actors/builtin/codes.go]: Thanks. I implemented it but then turns out all the call sites disappeared when it stopped being a union. I'll delete it.

**Pull request merged**